src/Symfony/Component/Serializer/Serializer.php

```diff
@@ -51,46 +51,42 @@ class Serializer implements SerializerInterface, NormalizerInterface, Denormaliz
      */
     public const EMPTY_ARRAY_AS_OBJECT = 'empty_array_as_object';
 
-    private const SCALAR_TYPES = [
-        'int' => true,
-        'bool' => true,
-        'float' => true,
-        'string' => true,
-    ];
+    private NormalizerInterface $normalizer;
+
+    private DenormalizerInterface $denormalizer;
 
     protected ChainEncoder $encoder;
 
     protected ChainDecoder $decoder;
 
-    /**
-     * @var array<string, array<string, array<bool>>>
-     */
-    private array $denormalizerCache = [];
-
-    /**
-     * @var array<string, array<string, array<bool>>>
-     */
-    private array $normalizerCache = [];
-
     /**
      * @param array<NormalizerInterface|DenormalizerInterface> $normalizers
      * @param array<EncoderInterface|DecoderInterface>         $encoders
      */
     public function __construct(
-        private array $normalizers = [],
+        array $normalizers = [],
         array $encoders = [],
+        ChainNormalizerInterface $normalizerAggregator = null,
+        ChainDenormalizerInterface $denormalizerAggregator = null,
```

Comment:
```suggestion
        ChainDenormalizerInterface $denormalizerAggregator = new ChainDenormalizer(),
```

Reply:
That is a good idea. But I cannot do this now when we removed `ChainDenormalizerInterface` interface. (I need to throw exception if this is not null and you also pass an array of $normalizers.